service/src/main/java/io/camunda/service/UserTaskServices.java

```diff
@@ -123,17 +164,17 @@ public CompletableFuture<UserTaskRecord> updateUserTask(
   }
 
   public UserTaskEntity getByKey(final long userTaskKey) {
+    final var query = UserTaskQuery.of(q -> q.filter(f -> f.userTaskKeys(userTaskKey)));
     final var result =
-        userTaskSearchClient
-            .withSecurityContext(securityContextProvider.provideSecurityContext(authentication))
-            .searchUserTasks(userTaskSearchQuery(q -> q.filter(f -> f.userTaskKeys(userTaskKey))));
-    final var userTaskEntity = getSingleResultOrThrow(result, userTaskKey, "User task");
+        search(query, securityContextProvider.provideSecurityContext(authentication));
+
+    final var userTask = getSingleResultOrThrow(result, userTaskKey, "User task");
     final var authorization = Authorization.of(a -> a.processDefinition().readUserTask());
     if (!securityContextProvider.isAuthorized(
-        userTaskEntity.processDefinitionId(), authentication, authorization)) {
+        userTask.processDefinitionId(), authentication, authorization)) {
       throw new ForbiddenException(authorization);
     }
```

Comment:
🔧 With the introduction of `getSingleResultOrThrow` and including the permission definition in the private `search` method, you can remove the auth check here. We will run into NOT_FOUND if permissions are missing.